framework/src/org/checkerframework/framework/type/BoundsInitializer.java

```diff
@@ -47,6 +50,71 @@
     //==================================================================================================================
 
     /**
+     * Initializes the type arguments of {@code declaredType}. The upper bound of unbound wildcards
+     * is set to the upper bound of the type parameter for which it is an argument. If {@code
+     * declaredType} is raw, then the type arguments are uninferred wildcards.
+     *
+     * @param declaredType type whose arguments are initialized.
+     */
+    public static void initializeTypeArgs(AnnotatedDeclaredType declaredType) {
+        final DeclaredType actualType = (DeclaredType) declaredType.actualType;
+        if (actualType.getTypeArguments().isEmpty() && !declaredType.wasRaw()) {
+            // No type arguments to infer.
+            return;
+        }
+
+        final TypeElement typeElement =
+                (TypeElement) declaredType.atypeFactory.types.asElement(actualType);
+        final List<AnnotatedTypeMirror> typeArgs = new ArrayList<>();
+
+        for (int i = 0; i < typeElement.getTypeParameters().size(); i++) {
+            TypeMirror javaTypeArg =
+                    getJavaTypeArg(declaredType, typeElement.getTypeParameters(), i);
+            final AnnotatedTypeMirror typeArg =
+                    AnnotatedTypeMirror.createType(javaTypeArg, declaredType.atypeFactory, false);
+            switch (typeArg.getKind()) {
+                case WILDCARD:
+                    AnnotatedWildcardType wildcardType = (AnnotatedWildcardType) typeArg;
+                    wildcardType.setTypeVariable(typeElement.getTypeParameters().get(i));
+                    if (declaredType.wasRaw()) {
+                        wildcardType.setUninferredTypeArgument();
+                    }
+
+                    initializeExtendsBound(wildcardType);
+                    initializeSuperBound(wildcardType);
+                    break;
+                case TYPEVAR:
+                    initializeBounds((AnnotatedTypeVariable) typeArg);
+                    break;
+                default:
+                    // do nothing
+            }
+            typeArgs.add(typeArg);
+        }
+
+        declaredType.typeArgs = Collections.unmodifiableList(typeArgs);
+    }
+
+    /**
+     * Returns the underlying java type of the {@code i} type parameter of {@code type}. If {@code
```

Comment:
The `type parameter` should be `type argument` to match the method name.
Changing it to `{@code i}-th` makes it a bit easier to read.
Same comments for `@return` below.